### DISCUSSION TITLE
Tasmota for Tuya WRG1 Module (RTL8711AM)

### DISCUSSION THREAD

**Discussion by kirovilya** [arendst/Tasmota#9262] [Ideas]
**Have you looked for this feature in other issues and in the docs?**  
yes

**Is your feature request related to a problem? Please describe.**  
_A clear and concise description of what the problem is._
I have not found Tasmota firmware for Tuya WRG1 Module. 

**Describe the solution you'd like**  
_A clear and concise description of what you want to happen._
I would like to use a cheap zigbee gateway in a local network (without cloud and Tuya App). Same as with Sonoff Zigbee Bridge with Tasmota firmware.

**Describe alternatives you've considered**  
_A clear and concise description of any alternative solutions or features you've considered._
Sonoff Zigbee Bridge with Tasmota firmware.

**Additional context**  
_Add any other context or screenshots about the feature request here._
Tuya wireless Hub (https://aliexpress.ru/item/4000465421911.html)

![photo_2020-09-07_19-28-49 (2)](https://user-images.githubusercontent.com/8360230/92440373-14e2bd80-f1b5-11ea-8c83-85ee2966a9c8.jpg)
![photo_2020-09-07_19-28-49](https://user-images.githubusercontent.com/8360230/92440379-1613ea80-f1b5-11ea-9395-7685ef267067.jpg)
![Снимок экрана от 2020-09-08 09-16-58](https://user-images.githubusercontent.com/8360230/92440380-1613ea80-f1b5-11ea-9415-f091cbb21489.png)
![IMG_20200907_202521](https://user-images.githubusercontent.com/8360230/92440384-16ac8100-f1b5-11ea-9e4e-9ace3b492e44.jpg)
![IMG_20200907_202513](https://user-images.githubusercontent.com/8360230/92440386-17451780-f1b5-11ea-917f-3cd59c70d847.jpg)

Datasheet for WRG1 module https://developer.tuya.com/en/docs/iot/device-development/module/wifi-module/wrg1-datasheet

A similar issue https://github.com/arendst/Tasmota/issues/8583

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
As long there is no full Arduino RTL8711 support this will not be possible since Tasmota needs the Arduino layer

**Comment by sfromis** [comment]
For the foreseeable future, the only way of getting Tasmota on that device would be to replace the WRG1 module with something including a supported chip, which may need a birds nest of wiring to get pin compatibility.

**Comment by Jason2866** [comment]
Changing the WIFI module is not enough. The zigbee module is not supported. It is a different chip as in the zigbeebridge.
There will be no support for this device

**Comment by Stoatwblr** [comment]
It looks like realtek might be helping out: 

https://www.amebaiot.com/en/ameba-arduino-getting-started-rtl8710/

**Comment by ascillato** [comment]
Good news!!!!!

Arduino Core for RT wifi SoC: https://github.com/Ameba8195/Arduino

Arduino Board Package: https://github.com/Ameba8195/Arduino/raw/master/release/package_realtek.com_ameba_index.json

https://www.cnx-software.com/2021/01/10/rtlduino-rtl8720dn-dual-band-wifi-iot-board-features-2-4-5ghz-wireless-mcu/amp/

**Comment by Jason2866** [comment]
@ascillato  Now it needs "someone" to build the PlatformIO package out of it
Maybe ? https://github.com/platformio/platformio-core/issues/3639#issuecomment-691492703

**Reply by Stoatwblr** [reply]
paging @realtekycchang7 and @Ameba8195

**Comment by charlesbrumbaugh** [comment]
I stumbled on this and I would also love to see Tasmota for Tuya WRG1 Module (RTL8711AM).  Any update on this?

**Reply by sfromis** [reply]
No, and not likely to come. I'd not expect it ever.

**Comment by Jason2866** [comment]
Time is running against RTL8711. The Arduino port is still very basic. No PlatformIO port...
Tasmota will probably ported to ESP32-C3. Espressif is investing development work in it 
and the changes to port Tasmota will probably not that big

**Reply by faizalps501** [reply]
any good news :-)

**Reply by sfromis** [reply]
If by that you mean Tasmota getting to run on RTL8711, no.

**Reply by faizalps501** [reply]
I have zigbee hub with WRG1. I would like to make it connect with home assistant as a zigbee coordinator . 
Is there is anyway other than tasmotize ? Or tasmotize is possible with this module ?

**Reply by ascillato** [reply]
@faizalps501 Sorry, but as explained in this thread, Tasmota is not supported in RTL8711 chips, so it is not possible to do what you are asking.

**Reply by sfromis** [reply]
And even if you choose to replace the WRG1 module with one supported by Tasmota, the TYZS3 module is not supported by Tasmota either. Whether you can get it connected to HA in some other way is not a question relevant to Tasmota, which is the topic here. It does not cost much money to get a coordinator supported by Tasmota....

**Comment by Drewix** [comment]
Is there any new information/someone succeeded in changing the software from tuya to ESP home, tasmote or OpenBeken?

**Reply by barbudor** [reply]
Tasmota is still a definitive NO
For the other software I suggest you ask on their respective forum to increase your chance to get an answer

**Reply by parasite85** [reply]
See: https://github.com/parasite85/tuya_tygwzw1_hack

**Comment by parasite85** [comment]
@kirovilya: maybe this will be intresting for you [https://github.com/parasite85/tuya_tygwzw1_hack](https://github.com/parasite85/tuya_tygwzw1_hack).

I have no time to write tasmota extensions or similar stuff to tasmota but this might help you.

**Reply by kirovilya** [reply]
@parasite85 Thank you. I'll take a look. True, I have already removed this gateway far away :)

**Comment by RneeJ** [comment]
bump, new info ? : 
Tuya developer website : 
About Arduino : https://developer.tuya.com/en/docs/iot/arduino-implements-simple-demo?id=K96yyxgetltzj
About WRG 1 : https://developer.tuya.com/en/docs/iot/wrg1-datasheet?id=K97rig6mscj8e

Library : 
https://github.com/tuya/tuya-wifi-mcu-sdk-arduino-library
https://github.com/libretiny-eu/libretiny

This router (TYZG1 = € 7,00) includes RTL8711AM and has nice connectors : 
![afbeelding](https://github.com/user-attachments/assets/e067daa6-a3a5-4db0-8ce5-6fbb71d5c7c3)
![afbeelding](https://github.com/user-attachments/assets/03871e29-532f-4633-b183-8be518b440ee)

**Reply by sfromis** [reply]
No news, dont expect Tasmota to run on that chip, ever.

**Reply by RneeJ** [reply]
to bad, thank you for your quick response.

**Comment by divadiow** [comment]
[Converting WRG1 and TYZS3 Zigbee Gateways to OpenBeken](https://www.elektroda.com/rtvforum/topic4130211.html)

**Reply by RneeJ** [reply]
thank you, looks very promising !
